flink-runtime/src/main/java/org/apache/flink/streaming/api/connector/sink2/StandardSinkTopologies.java

```diff
@@ -39,12 +41,29 @@ private StandardSinkTopologies() {}
      */
     public static <CommT> void addGlobalCommitter(
             DataStream<CommittableMessage<CommT>> committables,
-            SerializableSupplier<Committer<CommT>> committerFactory,
+            SerializableFunction<CommitterInitContext, Committer<CommT>> committerFactory,
             SerializableSupplier<SimpleVersionedSerializer<CommT>> committableSerializer) {
         committables
                 .getExecutionEnvironment()
                 .addOperator(
                         new GlobalCommitterTransform<>(
                                 committables, committerFactory, committableSerializer));
     }
+
+    /**
+     * Adds a global committer to the pipeline that runs as final operator with a parallelism of
+     * one.
+     */
+    public static <CommT> void addGlobalCommitter(
```

Comment:
Nit: Should we mark this method as deprecated to signal it will eventually be removed?

Reply:
It's fine to keep if the global committer doesn't need access to the ctx. It's relatively little code and could be considered sugar.